ethereum/src/BridgingPayment.sol

```diff
@@ -0,0 +1,71 @@
+pragma solidity ^0.8.24;
+
+abstract contract BridgingPayment {
+    event FeePaid();
+
+    error NotAnAdmin();
+    error NotEnoughFunds();
+
+    address public underlying;
+
+    uint256 fee;
+    address admin;
+
+    constructor(address _underlying, address _admin, uint256 _fee) payable {
+        underlying = _underlying;
+        admin = _admin;
+        fee = _fee;
+    }
+
+    /** @dev Deduct a `fee` from the user and transfer it to the `admin` address.
+     *  This function reverts if user don't have enough funds to pay the fee.
+     */
+    function deductFee() internal {
+        (bool feeTransferSuccess, ) = admin.call{value: fee}("");
```

Comment:
For sending Ether here, I would use transfer:
https://dev.to/tchisom17/understanding-ether-transfers-in-solidity-send-transfer-and-call-3k67#:~:text=Your%20smart%20contract%20in%20Solidity,to%20handle%20transfer%20failures%20manually.